hack/generated/pkg/reflecthelpers/reflect_helpers.go

```diff
@@ -208,3 +201,40 @@ func FindResourceReferences(transformer genruntime.ARMTransformer) (map[genrunti
 
 	return result, nil
 }
+
+// TODO: This method is a bit busy...
```

Comment:
This function doesn't really hold together - each of the 4 sections of the function is using a different input, which is why it needs 5 args. Maybe just put a function around the scope handling at the end and merge the rest back into `ConvertResourceToDeployableResource`? 

If you want that function to be shorter, it would make sense to move the reflection code lines 26-45 into a method like `MetaObject.AsARMTransformer()` - then everything in `ConvertResourceToDeployableResource` would be working at the same abstraction level, I think.

Reply:
but this function is building something from those 5 (really 4, because `ctx` can kinda be ignored) inputs. Those are the 4 inputs required to make a `genruntime.ConvertToARMResolvedDetails`. The awkward/annoying thing is that:
1. We start with `metaObject`.
2. We create `armTransformer` and `resourceHierarchy` from `metaObject`.
3. Neither `armTransformer` nor `resourceHierarchy` "contains" the original `metaObject` though, so we end up needing to pass all 3 around. Even in `CR2DR` all 3 of those types are used later on in the method (although `metaObject` is just for `metaObject.GetName()` used in logging which we could probably come up with an alternative way to do).

Pulling all of this back into `CR2DR` feels worse than having it pulled out (although I'll be the first to admit I don't love how I factored it out).

What we really have here is `metaObject` and then "data derived from or associated with `metaObject`" that I kinda want to pass around together...

Would this pattern be better (looking at what we do in CR2DR)?
1. Resolve what we need to resolve (`resourceHierarchy`, `resolved genruntime.ConvertToARMResolvedDetails`)
2. Extract `armTransformer`.
3. Call `armTransformer.ConvertToARM(resolved)`
4. Do the stuff at the bottom (under the comment): "// We have different deployment models for Subscription rooted vs ResourceGroup rooted resources"

I've sent an update to this PR that does this (in a fresh commit so you can just look at it if you like). Let me know if you think it's better.

Reply:
Taking a look - I'll do another review so I can tie comments to code lines.